### PR TITLE
Updated some missing params and return sections in the documentation.

### PR DESCRIPTION
**Pull request by nomad-software**
Added some missing parameters and return sections to the documentation.

In `std.conv` i've moved part of the doc block and the unittest to better document the `to` function within the `castFrom` template, a'la [`std.algorithm.aggregate`](http://dlang.org/phobos/std_algorithm_iteration.html#aggregate).

All changes can be previewed here: http://nomad.so/misc/d/dlang.org/fix-ddoc-parameter-warnings/phobos-prerelease/index.html

### CODE REVIEW

File: std/string.d [hunk 1]
```diff
@@ -1966,10 +1972,16 @@ auto representation(Char)(Char[] s) @safe pure nothrow @nogc
 }
 
 
-/++
-    Capitalize the first character of $(D s) and convert the rest of $(D s)
-    to lowercase.
- +/
+/**
+ * Capitalize the first character of $(D s) and convert the rest of $(D s)
```
Comment (MetaLang) [inline std/string.d]: This only works for characters in the ASCII range, right? If so, you should mention that.

**Comment by MetaLang** [comment]
Doc LGTM minus capitalize

File: std/string.d (hunk 1 above)
Comment (nomad-software) [inline std/string.d]: As far as I can see it supports unicode too.

Comment (MetaLang) [inline std/string.d]: You're write, it does. I was on mobile and couldn't view the whole file. Never mind me.

File: std/conv.d [hunk 2]
```diff
@@ -5275,15 +5275,18 @@ unittest
     Params:
         From  = The type to cast from. The programmer must ensure it is legal
                 to make this cast.
-        To    = The type to cast to
-        value = The value to cast. It must be of type $(D From),
-                otherwise a compile-time error is emitted.
-
-    Returns:
-        the value after the cast, returned by reference if possible
```
Comment (schuetzm) [inline std/conv.d]: Not sure about this. The template and its member are always to be used together. It could be confusing if the documentation is split. Have you tested how this appears in the generated documentation?

Comment (nomad-software) [inline std/conv.d]: This actually aids readability. Please read the initial pull description as there is a similar case and a link to a preview of the generate documentation.

Comment (schuetzm) [inline std/conv.d]: Ok, it's not ideal, but `to` was previously completely undocumented, it's definitely better now. LGTM then.

File: std/string.d [hunk 3]
```diff
@@ -1918,6 +1918,12 @@ ptrdiff_t lastIndexOfNeither(Char,Char2)(const(Char)[] haystack,
  * Returns the representation of a string, which has the same type
  * as the string except the character type is replaced by $(D ubyte),
  * $(D ushort), or $(D uint) depending on the character width.
+ *
+ * Params:
+ *     s = The string to return the representation of.
```
Comment (JakobOvrum) [inline std/string.d]: s/representation/_representation/

File: std/string.d [hunk 4]
```diff
@@ -1918,6 +1918,12 @@ ptrdiff_t lastIndexOfNeither(Char,Char2)(const(Char)[] haystack,
...
+ *
+ * Returns:
+ *     The representation of the passed string.
```
Comment (JakobOvrum) [inline std/string.d]: ditto

File: std/string.d [hunk 5]
```diff
@@ -1966,10 +1972,16 @@ auto representation(Char)(Char[] s) @safe pure nothrow @nogc
...
-/++
-    Capitalize the first character of $(D s) and convert the rest of $(D s)
-    to lowercase.
- +/
+/**
+ * Capitalize the first character of $(D s) and convert the rest of $(D s)
+ * to lowercase.
+ *
+ * Params:
+ *     s = The string to capitalize.
```
Comment (JakobOvrum) [inline std/string.d]: s/capitalize/_capitalize/

File: std/bitmanip.d [hunk 6]
```diff
@@ -3191,6 +3191,7 @@ unittest
         T     = The integral type to convert the first $(D T.sizeof) bytes to.
         endianness = The endianness to write the bytes in.
         range = The range to write to.
+        value = The value to write.
```
Comment (JakobOvrum) [inline std/bitmanip.d]: s/value/_value/

File: std/bitmanip.d [hunk 7]
```diff
@@ -3536,6 +3537,7 @@ unittest
         T     = The integral type to convert the first $(D T.sizeof) bytes to.
         endianness = The endianness to write the bytes in.
         range = The range to append to.
+        value = The value to append.
```
Comment (JakobOvrum) [inline std/bitmanip.d]: s/append/_append/

The above existing line also makes the same mistake (this is such a misfeature of DDoc...)

Comment (nomad-software) [inline std/bitmanip.d]: What does the underscore do?

Comment (JakobOvrum) [inline std/bitmanip.d]: DDoc for a symbol `foo` highlights `foo` in the documentation description, assuming that every use of the word `foo` is a reference to the symbol itself. The same is done for parameter names. The underscore tells DDoc that the word is used in its original meaning, not as a reference to a symbol.

Comment (nomad-software) [inline std/bitmanip.d]: Right, I get it, i'll amend it.

Comment (nomad-software) [inline std/bitmanip.d]: Looking at the output, the parameter names are not highlighted in the same way as symbol names are so this might be a unnecessary change for parameter descriptions.

**Comment by nomad-software** [comment]
Updated documentation to remove unnecessary symbol highlights in descriptions. I've updated the preview to, see the link in the original pull description.

**Comment by JakobOvrum** [comment]
Auto-merge toggled on
